tests/test_main.py

```diff
@@ -1,6 +1,324 @@
-# TODO: add tests
-from app.main import hello_world
+import inspect
+import io
+from contextlib import redirect_stdout
 
+import pytest
 
-def test_hello_world():
-    assert hello_world() == "Hello, world!"
+from app.main import Player, Elf, Dwarf, ElfRanger, Druid, DwarfWarrior, DwarfBlacksmith, calculate_team_total_rating, \
+    elves_concert, feast_of_the_dwarves
+
+
+@pytest.mark.parametrize(
+    "class_,methods",
+    [
+        (Player, ["get_rating", "player_info"]),
+        (Elf, ["get_rating", "player_info", "play_elf_song"]),
+        (ElfRanger, ["get_rating", "player_info", "play_elf_song"]),
+        (Druid, ["get_rating", "player_info", "play_elf_song"]),
+        (Dwarf, ["get_rating", "player_info", "eat_favourite_dish"]),
+        (DwarfWarrior, ["get_rating", "player_info", "eat_favourite_dish"]),
+        (DwarfBlacksmith, ["get_rating", "player_info", "eat_favourite_dish"]),
+    ],
+)
+def test_classes_should_have_corresponding_methods(class_, methods):
+    for method in methods:
+        assert (
+                hasattr(class_, method) is True
```

Comment:
When we are not in CodeEditor tests - this `is True` is unnecessary - and IIRC - bad practice